taxi/views.py

```diff
@@ -61,10 +63,24 @@ class CarListView(LoginRequiredMixin, generic.ListView):
 class CarDetailView(LoginRequiredMixin, generic.DetailView):
     model = Car
 
+    def post(self, request, *args, **kwargs):
+        car = self.get_object()  # Récupère l'objet Car via l'URL
+
+        # Vérifie si l'utilisateur est déjà conducteur de cette voiture
+        if request.user in car.drivers.all():
+            # Si l'utilisateur est déjà conducteur, on le supprime
+            car.drivers.remove(request.user)
+        else:
+            # Sinon, on l'ajoute à la liste des conducteurs
+            car.drivers.add(request.user)
+
+        # Redirection vers la page de détail de la voiture
+        return redirect(reverse("taxi:car-detail", kwargs={"pk": car.pk}))
```

Comment:
The `post` method in `CarDetailView` is correctly implemented to toggle the current user as a driver for the car. Ensure that this logic is intended for the detail view, as it might be more appropriate in a separate view dedicated to managing driver assignments.